test/libsolidity/semanticTests/viaYul/conversion/function_cast.sol

```diff
@@ -0,0 +1,14 @@
+contract C {
+    function f(uint x) public pure returns (uint) {
+        return 2 * x;
+    }
+    function g() public view returns (function (uint) external returns (uint)) {
+        return this.f;
+    }
+    function h(uint x) public returns (uint) {
+        return this.g()(x) + 1;
+    }
+}
+// ----
```

Comment:
Might it be reasonable to compile this via Yul, if it is to serve as test for this ;-)?

Reply:
Also you could use an even simpler test for just testing the cast itself like, but I'm actually not sure that's better, but maybe doesn't hurt to just add it as well:
```
  function t() external view returns (function() external a, function() external view b) {
    a = this.f; b = this.f;
  }
```